var/spack/repos/builtin/packages/aluminum/package.py

```diff
@@ -37,13 +37,18 @@ class Aluminum(CMakePackage, CudaPackage):
             ' communication of accelerator data')
     variant('cuda_rma', default=False, description='Builds with support for CUDA intra-node '
             ' Put/Get and IPC RMA functionality')
+    variant('rccl', default=False, description='Builds with support for NCCL communication lib')
 
     depends_on('cmake@3.17.0:', type='build')
     depends_on('mpi')
     depends_on('nccl', when='+nccl')
     depends_on('hwloc@1.11:')
     depends_on('hwloc +cuda +nvml', when='+cuda')
     depends_on('cub', when='@:0.1,0.6.0: +cuda ^cuda@:10.99')
+    depends_on('cub', when='@:0.1,0.6.0: +rocm')
+
+    conflicts('+cuda', when='+rocm')
+    conflicts('+rocm', when='+cuda')
```

Comment:
```suggestion
    conflicts('+cuda', when='+rocm', msg='CUDA and ROCm support are mutually exclusive')
```
One `conflicts` is enough, and it can't hurt to add an error message :)